README.md

```diff
@@ -20,94 +21,140 @@ You should not use it for anything serious.**
+
+## Setup
+
+We provide a development Docker container with all of the dependencies you need
+for building the project, so the minimum requirements for getting started are
+[Docker](https://docs.docker.com/get-docker/) and git. For development, you'll
+also need the [ctez](https://github.com/tezos-checker/ctez) git submodule. To
+fetch all submodules, run:
 
-For development, you'll also need the [ctez](https://github.com/tezos-checker/ctez)
-submodule. To fetch all submodules, run:
 ```console
 $ git submodule update --init
 ```
 
-Within a `nix-shell` (the first time this might take a while, since it must
-fetch all dependencies), type
+With that out of the way you can start up the dev container. We use
+[Earthly](https://earthly.dev/) to specify and build our Docker images. We
+recommend launching the dev container as follows. This gives earthly access to
+your local Docker daemon for executing builds.
 
-* `make build-ocaml` to build and compile the ocaml code (in [./src](./src)).
-* `make generate-ligo` to generate the ligo code (in `./generated/ligo`).
-* `make build-ligo` to generate the michelson code (in `./generated/michelson`).
-* `make build` to do all of the above.
-* `make test` to run all the (OCaml) tests ([./src/tests.ml](./src/tests.ml)). Note that this takes several minutes.
-* `make fast-test` to run the fast (OCaml) tests
-* `make` to do all the above.
+```console
+docker run -it \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -v $PWD:/checker \
+  ghcr.io/tezos-checker/checker/dev
+```
 
-For test coverage report using bisect_ppx, type
-*  `make test-coverage` (report in `./_coverage/index.html`), or
-*  `make test-coverage.json` (report of per-file summary in `./test-coverage.json`).
+The rest of this guide assumes that you are running commands in the dev container.
 
-For extracting (haddock-style) documentation from the code using dune, type
-*  `make docs` (docs entrypoint: `./_build/default/_doc/_html/index.html`)
+Note that the dev container also includes all language-specific tooling, so you
+can run language-specific build commands as desired (e.g. `dune build .` or
+`poetry run <cmd>`). This can be helpful for ensuring that editor integrations
+function properly (e.g. `ocaml-platform` for VSCode).
+
+## Build
+
+All of the build targets specified in the project [Earthfile](./Earthfile) may
+be built within the dev container (or locally if you have earthly installed).
+Some helpful targets are:
+
+* `earthly +build-ocaml` to build and compile the OCaml code (in
+  [./src](./src)).
+* `earthly +build-ligo` to generate ligo and michelson code (produces local
+  output in `./generated`).
+* `earthly +test` to run the full test suite.
+
+For linting type:
+* `earthly +lint`
+
+For test coverage report using bisect_ppx, type:
+* `earthly +test-coverage` (report in `./_coverage/index.html`)
+
+For extracting (haddock-style) documentation from the code using dune, type:
+* `earthly --artifact +docs/ ocaml-docs/`
```

Comment:
Trying this locally creates a folder `tmp755184827` inside `ocaml-docs`; it might be easier to just invoke `SAVE ARTIFACT _build/default/_doc/_html AS LOCAL ocaml-docs` within the `+docs` target and just do `earthly +docs` to get the docs in `ocaml-docs` folder.